book/core_programming.md

```diff
@@ -1102,6 +1107,25 @@ def extended_methodology(data):
 
 ```
 
+```{code-tab} r R
+# some function that we want to keep closed for modification
+core_method <- function(data) {
+  ...
+  return(result)
+}
+
+
+# if we want to extend a function without modifying it, we can always do the following
+extended_methodology <- function(data){
+  core_results = core_method(data)
+  return(
+    extended_functionality(core_results)
```

Comment:
can you also define this function in the example?

Reply:
Fixed.